src/chat.rs

```diff
@@ -1040,7 +1040,7 @@ impl Chat {
         };
         let ephemeral_timestamp = match ephemeral_timer {
             EphemeralTimer::Disabled => 0,
-            EphemeralTimer::Enabled { duration } => timestamp + i64::from(duration),
+            EphemeralTimer::Enabled { duration } => time() + i64::from(duration),
```

Comment:
`timestamp` is "smeared", i.e. may be some seconds earlier or later in order to make it unique. We can't use it as in order to test it, it has to be exact (except if I make the test very generous and wait e.g. 5 seconds for the message to be deleted)